pkg/block/fetcher_test.go

```diff
@@ -393,3 +401,271 @@ func TestTimePartitionMetaFilter_Filter(t *testing.T) {
 	testutil.Equals(t, expected, input)
 
 }
+
+func TestCompactedBlockFilter_Filter(t *testing.T) {
+	for _, tcase := range []struct {
+		name     string
+		input    map[ulid.ULID]*metadata.Meta
+		expected func(input map[ulid.ULID]*metadata.Meta) map[ulid.ULID]*metadata.Meta
+	}{
+		{
+			name: "3 non compacted metas in bucket",
+			input: map[ulid.ULID]*metadata.Meta{
+				ULID(1): {
+					BlockMeta: tsdb.BlockMeta{
+						ULID: ULID(1),
+						Compaction: tsdb.BlockMetaCompaction{
+							Sources: []ulid.ULID{ULID(1)},
+						},
+					},
+				},
+				ULID(2): {
+					BlockMeta: tsdb.BlockMeta{
+						ULID: ULID(2),
+						Compaction: tsdb.BlockMetaCompaction{
+							Sources: []ulid.ULID{ULID(2)},
+						},
+					},
+				},
+				ULID(3): {
+					BlockMeta: tsdb.BlockMeta{
+						ULID: ULID(3),
+						Compaction: tsdb.BlockMetaCompaction{
+							Sources: []ulid.ULID{ULID(3)},
+						},
+					},
+				},
+			},
+			expected: func(input map[ulid.ULID]*metadata.Meta) map[ulid.ULID]*metadata.Meta {
+				return map[ulid.ULID]*metadata.Meta{
+					ULID(1): input[ULID(1)],
+					ULID(2): input[ULID(2)],
+					ULID(3): input[ULID(3)],
+				}
+			},
+		},
+		{
+			name: "compacted meta in bucket",
+			input: map[ulid.ULID]*metadata.Meta{
+				ULID(4): {
+					BlockMeta: tsdb.BlockMeta{
+						ULID: ULID(4),
+						Compaction: tsdb.BlockMetaCompaction{
+							Sources: []ulid.ULID{ULID(1), ULID(2), ULID(3), ULID(4)},
+						},
+					},
+				},
+			},
+			expected: func(input map[ulid.ULID]*metadata.Meta) map[ulid.ULID]*metadata.Meta {
+				return map[ulid.ULID]*metadata.Meta{
+					ULID(4): input[ULID(4)],
+				}
+			},
+		},
+		{
+			name: "two blocks with same sources",
+			input: map[ulid.ULID]*metadata.Meta{
+				ULID(3): {
```

Comment:
Configuring test cases should be clear and simple. I think we don't need to have the whole `metadata.Meta` We can just configure as input: `map[ulid.ULID]tsdb.BlockMetaCompaction` and that's it. We don't really configure anything else. Then we can create a proper map to pass to filter later on. 

However, piece with test cases should be concise and without many repetitions.

Reply:
not addressed